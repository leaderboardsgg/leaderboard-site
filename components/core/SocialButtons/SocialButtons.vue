<template>
  <div class="social-button--container">
    <a
      v-for="social in socials"
      :key="social.name"
      class="social-button"
      :alt="social.name"
      :href="social.url"
    >
      {{ social.icon }}
    </a>
  </div>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api';

export default defineComponent({
  props: {
    socials: {
      required: true,
      type: Array,
    },
  },
});
</script>

<style scoped>
.social-button--container {
  @apply my-1;
  @apply grid auto-cols-fr grid-cols-4 gap-3;
}

.social-button {
  @apply flex content-center items-center justify-center justify-items-center;
  @apply rounded bg-gray-100;
  @apply p-2;
}
</style>
