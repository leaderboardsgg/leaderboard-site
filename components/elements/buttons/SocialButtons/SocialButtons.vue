<script setup lang="ts">
interface NavLinksProps {
  socials: Socials[]
}

defineProps<NavLinksProps>()
</script>

<template>
  <div class="social-buttons">
    <a
      v-for="social in socials"
      :key="social.name"
      class="social-button"
      :alt="social.name"
      :href="social.url"
    >
      <img :src="`/icons/social/${social.icon}.svg`" :alt="social.icon" />
    </a>
  </div>
</template>

<style lang="postcss" scoped>
.social-buttons {
  @apply w-full grid auto-cols-fr grid-cols-4 gap-2;
}

.social-button {
  @apply flex content-center items-center justify-center;
  @apply rounded bg-gray-100;
  @apply p-2;
}

.social-button img {
  @apply w-5;
}
</style>
