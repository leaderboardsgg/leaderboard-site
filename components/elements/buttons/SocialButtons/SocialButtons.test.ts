import { mountSuspended } from '@nuxt/test-utils/runtime'
import SocialButtons from './SocialButtons.vue'

describe('<SocialButtons />', () => {
  it('should render without crashing', async () => {
    const wrapper = await mountSuspended(SocialButtons, {
      props: {
        socials: [
          {
            icon: 'disc',
            name: 'discord',
            url: 'https://discord.gg/leaderboards.gg',
          },
          {
            icon: 'twit',
            name: 'twitter',
            url: 'https://twitter.com/bestofdyingtwit',
          },
        ],
      },
    })

    expect(wrapper.isVisible()).toBe(true)
  })
})
