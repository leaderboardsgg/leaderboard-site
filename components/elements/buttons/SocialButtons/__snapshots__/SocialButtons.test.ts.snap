// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialButtons /> renders correctly 1`] = `
<div
  class="social-button--container"
>
  <a
    alt="Discord"
    class="social-button"
    href="https://discord.com/"
  >
    
    dscrd
  
  </a>
  <a
    alt="Twitch"
    class="social-button"
    href="https://twitch.tv/johnsmithruns"
  >
    
    twch
  
  </a>
  <a
    alt="Twitter"
    class="social-button"
    href="https://twitter.com/johnsmithruns"
  >
    
    twtr
  
  </a>
</div>
`;
